internal/olm/operator/bundle/install.go

```diff
@@ -78,6 +78,12 @@ func (i *Install) setup(ctx context.Context) error {
 	i.OperatorInstaller.PackageName = labels["operators.operatorframework.io.bundle.package.v1"]
 	i.OperatorInstaller.CatalogSourceName = fmt.Sprintf("%s-catalog", i.OperatorInstaller.PackageName)
 	i.OperatorInstaller.StartingCSV = csv.Name
+	i.OperatorInstaller.SupportedInstallModes = operator.GetSupportedInstallModes(csv.Spec.InstallModes)
+
+	if i.OperatorInstaller.SupportedInstallModes.Len() == 0 {
+		return fmt.Errorf("operator %q is not installable: no supported install modes", csv.Name)
+	}
```

Comment:
Should not it be in a validation func instead?

Reply:
I added the check to the `CheckCompatibility` method and removed it from the setup.